src/main/java/io/javalin/core/JavalinConfig.java

```diff
@@ -164,6 +202,24 @@ public void server(Supplier<Server> server) {
 
     public static void applyUserConfig(Javalin app, JavalinConfig config, Consumer<JavalinConfig> userConfig) {
         userConfig.accept(config); // apply user config to the default config
+        Collection<Plugin> plugins = config.inner.plugins.values();
+
+        List<HandlerMetaInfo> registeredHandler = new ArrayList<>();
+        List<WsHandlerMetaInfo> registeredWsHandler = new ArrayList<>();
+        app.events(listener -> {
+            listener.handlerAdded(registeredHandler::add);
+            listener.wsHandlerAdded(registeredWsHandler::add);
+        });
+
+        plugins
+            .stream()
+            .filter(plugin -> plugin instanceof PluginLifecycleInit)
+            .forEach(plugin -> {
+                ((PluginLifecycleInit) plugin).init(app);
+                if (!registeredHandler.isEmpty() || !registeredWsHandler.isEmpty()) {
```

Comment:
`applyUserConfig` happens once inside `create(...)`, won't these lists always be empty? Or is this for if the plugin itself is adding routes?

Reply:
> Or is this for if the plugin itself is adding routes?

Exactly! I want to detect this to guarantee that the `init` method is called before any handlers are registered.

Reply:
There is a test for this if you want an example.

Reply:
Got it, looks good then!